src/osd/OSD.cc

```diff
@@ -8969,11 +8969,17 @@ void OSD::handle_fast_pg_create(MOSDPGCreate2 *m)
     epoch_t created = p.second.first;
     utime_t created_stamp = p.second.second;
     auto q = m->pg_extra.find(pgid);
-    if (q == m->pg_extra.end()) {
-      dout(20) << __func__ << " " << pgid << " e" << created
-               << "@" << created_stamp
-               << " (no history or past_intervals)" << dendl;
-      // pre-octopus ... no pg history.  this can be removed in Q release.
+    ceph_assert (q != m->pg_extra.end());
```

Comment:
How about we skip it and print a warning to clog and error?

Reply:
9aa772efa805fc7568114473c5bee5360be92d77